File: core/src/main/java/com/uiptv/util/AccountType.java
package com.uiptv.util;

import java.util.Arrays;

public enum AccountType {

    STALKER_PORTAL("Stalker Portal"),

    M3U8_LOCAL("M3U8 Local"),

    M3U8_URL("M3U8 URL"),

    XTREME_API("Xtreme API"),
    RSS_FEED("RSS Feed");

    private final String display;

    AccountType(String display) {
        this.display = display;
    }

    public String getDisplay() {
        return display;
    }

    public static AccountType getAccountTypeByDisplay(String display) {
        if (display == null || display.trim().isEmpty()) {
            throw new IllegalArgumentException("Account type display cannot be blank");
        }
        String normalizedDisplay = display.trim();
        return Arrays.stream(AccountType.values())
                .filter(a -> a.getDisplay().equalsIgnoreCase(normalizedDisplay))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Unknown account type display: " + display));
    }
}

File: server/src/main/kotlin/com/uiptv/util/AccountType.kt
package com.uiptv.util

enum class AccountType(private val display: String) {
    STALKER_PORTAL("Stalker Portal"),
    M3U8_LOCAL("M3U8 Local"),
    M3U8_URL("M3U8 URL"),
    XTREME_API("Xtreme API"),
    RSS_FEED("RSS Feed");

    fun getDisplay(): String = display

    companion object {
        @JvmStatic
        fun getAccountTypeByDisplay(display: String?): AccountType {
            require(!display.isNullOrBlank()) { "Account type display cannot be blank" }
            val normalizedDisplay = display.trim()
            return entries.firstOrNull { it.display.equals(normalizedDisplay, ignoreCase = true) }
                ?: throw IllegalArgumentException("Unknown account type display: $display")
        }
    }
}
